specs/commandline/plugin.md

```diff
@@ -41,51 +42,145 @@ Aliases:
+Upgrade a plugin
 
 Usage:
-  notation plugin remove [flags] <plugin name>
+  notation plugin upgrade [flags] <plugin_source>
 
 Flags:
-  -h, --help          help for remove
+  -h, --help                    help for install
+      --checksum string         must match SHA256 of the plugin source
+      --source string           the location of plugin installation file, options: "file", "url","registry" (default "file")                  
+```
 
+### notation plugin uninstall
+
+```text
+Uninstall a plugin
+
+Usage:
+  notation plugin uninstall [flags] <plugin_name>
+
+Flags:
+  -h, --help          help for remove
+  -y, --yes           do not prompt for confirmation
 Aliases:
-  remove, rm, uninstall, delete
+  remove, rm, uninstall
 ```
 
 ## Usage
 
-### Install a plugin
+## Install a plugin 
+
+### Install a plugin from file system
+
+Install a Notation plugin from file system and verify the plugin checksum.
 
 ```shell
-notation plugin install <plugin path>
+$ notation plugin install --source file <file_path> --checksum <digest>
 ```
 
-Upon successful execution, the plugin is copied to plugins directory and name+version of plugin is displayed. If the plugin directory does not exist, it will be created. When an existing plugin is detected, the versions are compared and if the existing plugin is a lower version then it is replaced by the newer version.
+If the user doesn't specify the `--source` flag, Notation will install the plugin from file system by default.
+
+```shell
+$ notation plugin install <file_path> --checksum <digest>
+```
+
+Upon successful execution, the plugin is copied to Notation's plugin directory. The name and version of the installed plugin is displayed as follows. 
+
+```console
+Successfully installed plugin <plugin name>, version <x.y.z>
+```
+
+If the plugin directory does not exist, it will be created. When an existing plugin is detected and the version is the same as the installing plugin, it fails to install and returns the error as follows. Users can use a flag `--force` to skip existence check and force the installation with a specified version.
+
+```console
+Error: failed to install the plugin, <plugin_name> already installed
+```
+
+If the entered plugin checksum digest doesn't match the published checksum, Notation will return an error message and will not start installation.
+
+```console
+Error: failed to install the plugin, input checksum does not match the published checksum, expected <digest>
+```
+
+### Install a plugin from URL
+
+Install a Notation plugin from a remote shared address and verify the plugin checksum.
+
+```shell
+$ notation plugin install --source url <URL> --checksum <digest>
+```
+
+### Install a plugin as an OCI artifact from a registry (for future iteration)
+
+Install a Notation plugin from a registry. Users can verify the plugin's signature with `notation verify` before the plugin installation.
+
+```shell
+$ notation plugin install --source registry <artifact_reference>
```

Comment:
Checksum check is always required:
```suggestion
$ notation plugin install --source registry --checksum <digest> <artifact_reference>
```

Reply:
Why an OCI image requires checksum? Maybe verify the signature and its image digest are enough to ensure the plugin's integrity and authenticity.

Reply:
That's why an example is required.

The `artifact_reference` contains the digest.